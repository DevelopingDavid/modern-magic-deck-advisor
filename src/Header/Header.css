.header {
  background-image: url(header-background.png);
  background-size: 100vw¡¡;
  background-repeat: no-repeat;
  display: flex; 
  justify-content: space-between;  
  height: 200px;
}

.magic-logo {
  height: 150px;
  margin: 30px;
  width: 600px;
}

.header--ul {
  list-style: none;
}

.header--ul li {
  margin: 20px;
}