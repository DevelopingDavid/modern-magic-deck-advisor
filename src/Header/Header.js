import React from 'react';
import './Header.css'

function Header() {
  return(
    <header className="header">
      <img src="./magic_logo.png" className="magic-logo"></img>
      <ul className="header--ul">
        <li>My Cards</li>
        <li>My Favorites</li>
        <li>My Wish List</li>
      </ul>
    </header>
  )
}

export default Header;