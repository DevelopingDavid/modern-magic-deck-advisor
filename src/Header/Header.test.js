import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';

const setAsideViewMock = jest.fn();
const setCardAreaViewMock = jest.fn();

describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Header
        setAsideView={setAsideViewMock}
        setCardAreaView={setCardAreaViewMock}/>
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  }); 

  it('should call setAsideView and setCardAreaView when My Cards is clicked', () => {
    wrapper.find('.my-cards').simulate('click');
    expect(setAsideViewMock).toBeCalled();
    expect(setCardAreaViewMock).toBeCalled();
  });

  it('should call setAsideView and setCardAreaView when Saved Decks is clicked', () => {
    wrapper.find('.fave-decks').simulate('click');
    expect(setAsideViewMock).toBeCalled();
    expect(setCardAreaViewMock).toBeCalled();
  });

  it('should call setAsideView and setCardAreaView when Wish List is clicked', () => {
    wrapper.find('.my-wish-list').simulate('click');
    expect(setAsideViewMock).toBeCalled();
    expect(setCardAreaViewMock).toBeCalled();
  });
}) ;