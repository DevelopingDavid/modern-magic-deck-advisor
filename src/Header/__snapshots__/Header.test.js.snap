// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<header
  className="header"
>
  <img
    alt="Magic The Gathering"
    className="magic-logo"
    src="./magic_logo.png"
  />
  <ul
    className="header--ul"
  >
    <li
      className="my-cards"
      onClick={[Function]}
    >
      My Cards
    </li>
    <li
      className="fave-decks"
      onClick={[Function]}
    >
      My Favorites
    </li>
    <li
      className="my-wish-list"
      onClick={[Function]}
    >
      My Wish List
    </li>
  </ul>
</header>
`;
